client/src/pages/admin-pages/addFlight/index.js

```diff
@@ -9,90 +9,82 @@ import {
   Button,
   Header,
   SideNav
-} from "../../components";
+} from "../../../components";
 
 import "./addflight.css";
 
 class AddFlight extends Component {
   state = {
     form: {},
-    selectedCity: {},
+    selectedValue: {},
     flightDuration: "",
     succesfullyAdded: false,
-    alert: null
+    alert: false,
+    error: false
   };
 
   handleAddFlight = e => {
     e.preventDefault();
-    const { dateValue, airCraftTypeValue } = this.state.form;
-    const { originValue, destinationValue } = this.state.selectedCity;
+
+    const {
+      flightDuration,
+      selectedValue: { originValue, destinationValue, airCraftTypeValue },
+      form: { dateValue, flightNumberValue }
+    } = this.state;
 
     const data = JSON.stringify({
+      flightDuration,
       originValue,
       destinationValue,
       dateValue,
-      airCraftTypeValue
+      airCraftTypeValue,
+      flightNumberValue
     });
-    fetch("/addFlight_post", {
+    fetch("/api/v1/flight", {
       credentials: "same-origin",
       headers: {
         "content-type": "application/json"
       },
       method: "POST",
       body: data
     })
-      .then(response => response.json(data))
+      .then(response => response.json())
       .then(data => {
         if (data.succesfullyAdded) {
           this.setState({
-            succesfullyAdded: true
+            succesfullyAdded: true,
+            alert: true
           });
-          this.handleSuceessAddFlight();
         } else {
           this.setState({
             succesfullyAdded: false
           });
         }
       })
       .catch(err => {
-        console.log("There has been an error ", err);
+        this.setState({
+          error: true
+        });
       });
   };
 
-  handleSuceessAddFlight() {
-    const getAlert = () => (
-      <SweetAlert
-        success
-        title="Done!"
-        confirmBtnBsStyle="success"
-        cancelBtnBsStyle="success"
-        onConfirm={() => this.hideAlert()}
-      >
-        New Flight is Added
-      </SweetAlert>
-    );
-
-    this.setState({
-      alert: getAlert()
-    });
-  }
-
-  hideAlert() {
+  hideAlert = () => {
     this.setState({
-      alert: null
+      alert: false
     });
-  }
+  };
 
   handleSelectChange = e => {
-    const { selectedCity } = this.state;
-    const updatedSelectedCity = { ...selectedCity };
-    updatedSelectedCity[e.target.name] = e.target.value;
-    this.setState({ selectedCity: updatedSelectedCity });
+    const { selectedValue } = this.state;
+    const { name, value } = e.target;
+    const updatedSelectedCity = { ...selectedValue };
+    updatedSelectedCity[name] = value;
```

Comment:
I think you'll need to declare updatedSelectedCity with `let`, not `const`, since you change it on line 81.
Another way to write this (and keep const) would be to use a dynamic key:
```
const updatedSelectedCity = {
  ...selectedValue,
  [name]: value
}
```

Reply:
This is neat and clean way to write this :+1: 

But I think we can edit the object declared with `const` internally. It will only throw an error if we try to re-assign the variable to a completely new object. Isn't that right? @tomduggan85

Reply:
I just tried, and you’re right! Sorry about that. Setting keys on a const object doesn’t throw.